### PR TITLE
:sparkles: Simplify usage of EnqueueRequestsFromMapFunc

### PR DESCRIPTION
**Pull request by alvaroaleman**
Makes creation of custom mappers a lot simpler

/assign @vincepri

### CODE REVIEW

File: pkg/handler/enqueue_mapped.go [hunk 1]
```diff
@@ -25,6 +25,22 @@ import (
...
+//
+// For UpdateEvents which contain both a new and old object, the transformation function is run on both
+// objects and both sets of Requests are enqueue.
+func FromMapFunc(mapFN func(MapObject) []reconcile.Request) EventHandler {
+	return &EnqueueRequestsFromMapFunc{
+		ToRequests: ToRequestsFunc(mapFN),
+	}
+}
+
 var _ EventHandler = &EnqueueRequestsFromMapFunc{}
```
Comment (alvaroaleman) [inline pkg/handler/enqueue_mapped.go:44]: Alternatively, we can completely unexport `EnqueueRequestsFromMapFunc`. It doesn't provide any advantage over using `FromMapFunc` and is a lot more annoying to construct

Comment (vincepri) [reply pkg/handler/enqueue_mapped.go:44]: +1 to that, better have a single way to use it than two and confuse users

Comment (vincepri) [reply pkg/handler/enqueue_mapped.go:44]: Let's make that a ⚠️

Comment (alvaroaleman) [reply pkg/handler/enqueue_mapped.go]: Updated, PTAL

File: pkg/handler/enqueue_mapped.go [hunk 2]
```diff
@@ -36,33 +34,41 @@ var _ EventHandler = &EnqueueRequestsFromMapFunc{}
 //
 // For UpdateEvents which contain both a new and old object, the transformation function is run on both
 // objects and both sets of Requests are enqueue.
-type EnqueueRequestsFromMapFunc struct {
+func EnqueueRequestsFromMapFunc(mapFN func(MapObject) []reconcile.Request) EventHandler {
```
Comment (vincepri) [inline pkg/handler/enqueue_mapped.go:37]:
```suggestion
func MapFunc(mapFN func(MapObject) []reconcile.Request) EventHandler {
```

Do we want to go for something a little simpler? wdyt?

Comment (alvaroaleman) [reply pkg/handler/enqueue_mapped.go:37]: All existing mappers are called `EnqueueRequestSomething` I thought it would be nice to keep that pattern

Comment (alvaroaleman) [reply pkg/handler/enqueue_mapped.go:37]: It also makes this change easier for existing users

**Comment by DirectXMan12** [comment]
FYI, this is breaking

**Pull request merged**